BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeGraphQLResponseParser.kt

```diff
@@ -0,0 +1,53 @@
+package com.braintreepayments.api
+
+import androidx.annotation.VisibleForTesting
+import org.json.JSONObject
+import java.net.HttpURLConnection
+
+/**
+ * Class that handles parsing http responses for [BraintreeGraphQLClient].
+ */
+internal class BraintreeGraphQLResponseParser @VisibleForTesting constructor(
+    private val baseParser: HttpResponseParser
+) : HttpResponseParser {
+
+    constructor() : this(BaseHttpResponseParser())
+
+    /**
+     * @param responseCode the response code returned when the http request was made.
+     * @param connection the connection through which the http request was made.
+     * @return the body of the http response.
+     */
+    @Throws(Exception::class)
+    override fun parse(responseCode: Int, connection: HttpURLConnection): String {
+        val response = baseParser.parse(responseCode, connection)
+        val errors = JSONObject(response).optJSONArray(GraphQLConstants.Keys.ERRORS)
+        if (errors != null) {
```

Comment:
Similar comment re: if we want to do something "guard-like" here in and return early